lib/teleterm/daemon/config.go

```diff
@@ -25,10 +27,27 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+type cluster interface {
+	GetServers(ctx context.Context) ([]clusters.Server, error)
+}
+
+type storage interface {
+	GetByURI(uri string) (cluster, error)
+}
+
+// This is used to wrap a real clusters.Storage so it meets our interface
+type StorageWrapper struct {
+	*clusters.Storage
+}
+
+func (sw *StorageWrapper) GetByURI(uri string) (cluster, error) {
+	return sw.Storage.GetByURI(uri)
+}
+
 // Config is the cluster service config
 type Config struct {
 	// Storage is a storage service that reads/writes to tsh profiles
-	Storage *clusters.Storage
+	Storage storage
```

Comment:
It is a bit awkward for an exported field in an exported struct type to have an unexported type. I would export `Storage` too.